commit 423869f8871dee4028b67fa30f660ab43d14cfb2
Author: Marc Zyngier <marc.zyngier@arm.com>
Date:   Tue Jan 29 15:19:23 2019 +0000

    irqchip/gic-v3-its: Gracefully fail on LPI exhaustion
    
    [ Upstream commit 45725e0fc3e7fe52fedb94f59806ec50e9618682 ]
    
    In the unlikely event that we cannot find any available LPI in the
    system, we should gracefully return an error instead of carrying
    on with no LPI allocated at all.
    
    Fixes: 38dd7c494cf6 ("irqchip/gic-v3-its: Drop chunk allocation compatibility")
    Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/irqchip/irq-gic-v3-its.c b/drivers/irqchip/irq-gic-v3-its.c
index d9a880108315..15579cba1a88 100644
--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -1581,6 +1581,9 @@ static unsigned long *its_lpi_alloc(int nr_irqs, u32 *base, int *nr_ids)
 		nr_irqs /= 2;
 	} while (nr_irqs > 0);
 
+	if (!nr_irqs)
+		err = -ENOSPC;
+
 	if (err)
 		goto out;
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1082_423869f8871dee4028b67fa30f660ab43d14cfb2.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1082_423869f8871dee4028b67fa30f660ab43d14cfb2.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
